### PR TITLE
[network_snapshot] Script to snapshot account information per epoch

### LINKED ISSUE #14: [monitor] all account state on each block/epoch
As requested by Sahil, we need to keep monitoring and keep track of all account state rewards earned by delegators.  Vlad had added an RPC call in this PR (https://github.com/harmony-one/harmony/pull/2254/files) to get all accounts in each block from the explorer node, and thus we can keep track of the account state.  Need to save the account info into a file or a db.

### PR DESCRIPTION
**Pull request by rhazberries**

### CODE REVIEW

File: network_snapshot/network_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+    """
+    log_formatter = logging.Formatter('%(message)s')
+    handler = logging.handlers.TimedRotatingFileHandler(log_file_path, when = 'h', interval = 2,
+                                                        backupCount = None, encoding = 'utf8')
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(log_formatter)
+    handler.rotator = _GZipRotator()
+    return handler
+
+def get_node_metadata(endpoint, retries):
```
Comment (LeoHChen) [inline network_snapshot/network_snapshot.py:57]: consider using the pyhmy package DM created.

File: network_snapshot/network_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+                prev_epoch = int(epoch_token)
+                # Since account filed generated, all blocks in that epoch should have been fetched
+                prev_block = get_epoch_first_block(int(epoch_token) + 1, blocks_per_epoch)
+            except Exception:
+                pass
+
+    # Create loggers after creating data directory
+    init_loggers()
+
+    while True:
```
Comment (LeoHChen) [inline network_snapshot/network_snapshot.py:470]: don't in the script, use a cronjob or a jenkins job to have a better control and test.

**Review by LeoHChen — COMMENTED** [review]
mostly okay, for RPC call, better put them into a library so that we can reuse them, like pyhmy or another pyhmylib.  For the while loop in main, better use cronjob to control it.

File: network_snapshot/network_snapshot.py (hunk 1 above)
Comment (rhazberries) [reply network_snapshot/network_snapshot.py:57]: The pyhmy lib uses the CLI to send requests, which is slow

**Pull request closed without merging**